### ISSUE TITLE
vardiff difficulty too low

### ISSUE THREAD

**Issue by calvintam236** [coinfoundry/miningcore#30]
I am using CPUs to mine XMR on `poolmining.org`, and the difficulty seems to be too low for my setup after vardiff kicks in.

This is one AMD Ryzen 1700. Diff dropped to 10000 from 30000 (default), so 1 share every ~10s..
```console
[2017-09-28 05:29:02] accepted (1/0) diff 30000 (187 ms)
[2017-09-28 05:29:05] speed 2.5s/60s/15m 547.1 549.2 n/a H/s max: 553.0 H/s
[2017-09-28 05:29:24] accepted (2/0) diff 30000 (183 ms)
[2017-09-28 05:29:35] accepted (3/0) diff 30000 (186 ms)
[2017-09-28 05:30:02] new job from xmr.poolmining.org:3032 diff 10000
[2017-09-28 05:30:05] speed 2.5s/60s/15m 550.4 550.3 n/a H/s max: 555.2 H/s
[2017-09-28 05:30:17] accepted (4/0) diff 10000 (187 ms)
[2017-09-28 05:30:22] accepted (5/0) diff 10000 (187 ms)
[2017-09-28 05:30:37] accepted (6/0) diff 10000 (181 ms)
[2017-09-28 05:31:05] speed 2.5s/60s/15m 550.0 550.1 n/a H/s max: 555.2 H/s
[2017-09-28 05:31:42] accepted (7/0) diff 10000 (185 ms)
[2017-09-28 05:31:52] accepted (8/0) diff 10000 (183 ms)
[2017-09-28 05:32:04] accepted (9/0) diff 10000 (188 ms)
```

**Comment by oliverw** [comment]
@shtse8 Could you have a look at this since you basically rewrote the vardiff stuff. 😃

**Comment by shtse8** [comment]
@oliverw sure.

@calvintam236 
could you paste your vardiff config here?

**Comment by calvintam236** [comment]
@shtse8 I am not running a pool. I am just a miner in a pool. Command I ran: `xmrig -a cryptonight -o xmr.poolmining.org:3032 -u {wallet_address}.{worker_name} --donate-level=1 --max-cpu-usage=100 --cpu-affinity=0xAAAA`.

**Comment by oliverw** [comment]
@shtse8 He's mining on my pool. I've just adjusted the vardiff config for monero so the issue should be solved since the default was simply too steep.

**Comment by oliverw** [comment]
@calvintam236 I've just checked with a competing pool and I think poolmining.org's default diff on the CPU mining port  is simply set too high for CPU

**Comment by calvintam236** [comment]
@oliverw It seems it is even lower now - difficulty is now 1000, and 1 share every ~1 second. Maybe your maximum VarDiff should be higher for CPU?

**Comment by oliverw** [comment]
Yeah that's strange. I see it in the logs right now. No idea why everyone is dropping to minDiff.

Config right now is:

```javascript
      "ports": {
        "3032": {
          "listenAddress": "0.0.0.0",
          "difficulty": 5000,
          "name": "CPU Mining",
          "varDiff": {
            "minDiff": 1000,
            "maxDiff": null,
            "targetTime": 15,
            "retargetTime": 90,
            "variancePercent": 30
          }
        },
        "3033": {
          "listenAddress": "0.0.0.0",
          "difficulty": 25000,
          "name": "GPU Mining",
          "varDiff": {
            "minDiff": 5000,
            "maxDiff": null,
            "targetTime": 15,
            "retargetTime": 90,
            "variancePercent": 30
          }
        },
        "3034": {
          "listenAddress": "0.0.0.0",
          "difficulty": 200000,
          "name": "NiceHash & Mining Farms",
          "varDiff": {
            "minDiff": 200000,
            "maxDiff": null,
            "targetTime": 15,
            "retargetTime": 90,
            "variancePercent": 30
          }
        }

```

**Comment by calvintam236** [comment]
Maybe `null` for `maxDiff` is not working? Can you try a fixed value?

**Comment by oliverw** [comment]
I've reverted the config to previous version. Your diff should return to at least 10k now.

**Comment by oliverw** [comment]
Current config

```javascript
"ports": {
  "3032": {
    "listenAddress": "0.0.0.0",
    "difficulty": 30000,
    "name": "CPU Mining",
    "varDiff": {
      "minDiff": 10000,
      "maxDiff": null,
      "targetTime": 15,
      "retargetTime": 90,
      "variancePercent": 30
    }
  },
  "3033": {
    "listenAddress": "0.0.0.0",
    "difficulty": 120000,
    "name": "GPU Mining",
    "varDiff": {
      "minDiff": 60000,
      "maxDiff": null,
      "targetTime": 15,
      "retargetTime": 90,
      "variancePercent": 30
    }
  },
  "3034": {
    "listenAddress": "0.0.0.0",
    "difficulty": 200000,
    "name": "High-End Multi GPU Mining",
    "varDiff": {
      "minDiff": 200000,
      "maxDiff": null,
      "targetTime": 15,
      "retargetTime": 90,
      "variancePercent": 30
    }
  }

```

**Comment by shtse8** [comment]
It's wired. the diff should increase if the average share submitted time is lower than the target time. and also will adjust the diff if lone time no share is submitted. So if the diff is too high, and no share can be submitted. the diff will be automatic lower. I have no ideas on this issue.

**Comment by oliverw** [comment]
Should fixed by 10b910f4f81ead34995bb42cce24fe0893550715. Re-open otherwise.
